### PR TITLE
Add acceptable and unacceptable conduct conflicts

### PR DESCRIPTION
**Pull request by watson02**
### Type of Change

- [ ] Feature
- [ ] Bug fix
- [x] Documentation

## Contributors

- @watson02 
- @mariakimheinert 
- @gvanzin-allegheny 
- @noorbuchi 
- @lukel43 

## Reminder

All GitHub Actions checks must be in a passing state in this branch or fork before this pull request can be merged.

### CODE REVIEW

File: conduct.md
Comment (noorbuchi) [inline conduct.md]: Can you please elaborate on this point. What do you mean by that? How does it apply in the context of the class interactions and conduct?

File: conduct.md
Comment (noorbuchi) [inline conduct.md]: I think rephrasing this sentence can increase its clarity. Some restructuring would be very helpful

File: conduct.md
Comment (noorbuchi) [inline conduct.md]: Again, I think some of these points could be elaborated on. Especially how they apply or how they are not acceptable. Please be as specific as you can

**Review by noorbuchi — CHANGES_REQUESTED** [review]
Thank you for your work on this. Please address the comments I made in my review. Additionally, please fix the linter and spelling issues in this document in order to pass the automated checks

File: conduct.md
Comment (lukel43) [reply conduct.md]: I like the idea, perhaps you could further highlight how students should work to prevent conflicts.

File: conduct.md
Comment (lukel43) [reply conduct.md]: Avoid one word sentences if possible. I like that the main idea seems to be having humility. Perhaps this could be elaborated on further.

**Review by gvanzin-allegheny — CHANGES_REQUESTED** [review]
Same things noor said.

**Review by gvanzin-allegheny — CHANGES_REQUESTED** [review]
There are a few spelling errors and markdown errors that github is flagging.  I'd fix those really quick and it should be good to push.  Besides that, the rest looks good to me.

**Review by mariakimheinert — CHANGES_REQUESTED** [review]
Hello @watson02! Please make sure all GitHub Actions are passing before requesting review.

**Comment by gvanzin-allegheny** [comment]
There are lint errors here:

Error: conduct.md:128:155 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
Error: conduct.md:134:25 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
Error: Failed with exit code: 1

And a spelling error here:

Warning: Unknown word (communiction)

**Review by lukel43 — CHANGES_REQUESTED** [review]
As previously mentioned, make sure the checks are passing then you should be good! Also you might want to find a better way to format it other than indenting the second line.

File: conduct.md
Comment (Yanqiao4396) [inline conduct.md]: Thanks for your hard work. Based on the issue tracker of the main file ( the one which is not in our fork), the subject is better to be "students" other than "you". Please consider to use "you"

File: conduct.md
Comment (noorbuchi) [inline conduct.md]: I really like points 2 and 3, they explain how we can prevent conflicts from starting altogether. But I'd like to see some elaboration or changes in point 1. There is no doubt that openness can help while working in a team, but how does it help specifically in the context of preventing conflicts **between group members**. This section is not really concerned with technical skills while coding.

File: conduct.md
Comment (noorbuchi) [inline conduct.md]: There is already a section concerning communication, so while this is a really good point, it might not fit here the way it's currently phrased. You could rephrase it to say something along the lines of how miscommunication and using inappropriate language escalates conflicts, which is why it's unacceptable.

**Review by noorbuchi — CHANGES_REQUESTED** [review]

File: conduct.md
Comment (noorbuchi) [inline conduct.md]: The sentence structure here is a bit weird. Humbleness and humility are synonymous. So saying that "Being humble is a form of humility" is kinda stating the obvious. Instead, you could rephrase it to something along the lines of this:

```suggestion
- Encourage humility: Being humble is an important aspect of working in a team to. It helps in understanding that it is not about one person but that it is about the team. 
```
You can also expand that to talk about how it helps in avoiding conflict if you'd like.

File: conduct.md
Comment (noorbuchi) [inline conduct.md]: I'm not entirely sure what you mean by this:
"One way to practice good communication is by giving good criticism which is a form of respect.". What is "good" criticism and how is it a form of respect?

Another note is, again, this point is more so concerning the communication section. In my opinion, it would be sufficient to just say this:
```suggestion
- Lack of Communication and miscommunication are primary causes of conflict. Abiding by the conducts described in the communication section prevents escalation and avoids issues from the start. 
```
However, I leave that up to you and the team to decide.

File: conduct.md
Comment (noorbuchi) [inline conduct.md]: Changing "not respecting others" to "not respectful of others" has a better flow in my opinion. What do you think?

**Review by noorbuchi — CHANGES_REQUESTED** [review]

File: conduct.md
Comment (mariakimheinert) [inline conduct.md]: All of these points contain great ideas! But, they don't conform to the format that the class agreed upon in [issue  #1](https://github.com/cmpsc-203-s22/code-of-conduct/issues/1). Specifically, they are not one sentence long and do not start with "Students...", as the class decided. Could they be reworded to follow this format? It would help the consistency throughout the conduct guide.

**Review by mariakimheinert — CHANGES_REQUESTED** [review]
Hi @watson02, I've left an overall suggestion below. Please let me know if you have any questions!

File: conduct.md
Comment (mariakimheinert) [inline conduct.md]: Each of these points should be one sentence long. Right now, they are all at least two sentences long with the short first sentence, e.g. "Resentment.". Could they be reworded so that the second sentence includes the concept introduced in the first such that the first is no longer needed?

File: conduct.md
Comment (mariakimheinert) [inline conduct.md]: These are all great ideas about how students should generally behave in the course. However, these acceptable and unacceptable conduct should relate specifically to conflict--they should describe what students should do if a conflict arises. For example, should students first try to resolve the conflict between themselves? What if the conflict escalates? Should they involve the instructor?

File: conduct.md
Comment (mariakimheinert) [inline conduct.md]: Similar to my comment above, this section should describe what students should not do if a conflict arises. For example, they probably shouldn't try to escalate the conflict or ignore it.

**Review by mariakimheinert — CHANGES_REQUESTED** [review]
@watson02, thanks for your work on these changes! I've made a few suggestions below. Please let me know if you have any questions about my comments!

**Review by gvanzin-allegheny — DISMISSED** [review]
The latest commit looks good to me, I believe you factored in the ideas professor wanted so I'll approve.  Also all checks passed so that's good as well.

File: conduct.md
Comment (Yanqiao4396) [inline conduct.md]: Using "a" instead of "an" in front of big.

File: conduct.md
Comment (Yanqiao4396) [inline conduct.md]: How about deleting "is" in front of  "case"

File: conduct.md
Comment (Yanqiao4396) [inline conduct.md]: "unresolved" instead of "unresovled"

**Review by Yanqiao4396 — CHANGES_REQUESTED** [review]
Most are good. I feel good about content. Please pay attention to some spell and grammar issues.

**Review by Yanqiao4396 — DISMISSED** [review]
Cool to me. Thanks for the hard work

**Review by gvanzin-allegheny — DISMISSED** [review]
Looks good.

**Review by lukel43 — DISMISSED** [review]
Fine with me! We can maybe update it further when trying to merge with the main class repository if needed.

**Comment by mariakimheinert** [comment]
@watson02, there are still a few of my comments that have not been addressed. Specifically, the format of these bullet points does not follow the format agreed upon by the class. Additionally, there are some issues with the content of these bullet points—they should be more specific to conduct in the context of conflicts.

It will be easier to address these requested changes in this PR, rather than after opening a PR to merge this fork into the main CoC repo since you would then need to create a new branch and PR to make the requested changes. So, I recommend that these changes be made in this PR.

**Comment by noorbuchi** [comment]
I agree with @mariakimheinert  on this one, please make sure to address the previous comments.

**Comment by watson02** [comment]
- Students should be open about their ablilties and limitations when coding.
- Students should retain other teammates ideas to let their teammates know that their opinions matter.
- Students should know that software engineering is a team effort and that it is not about one person but that it is about the whole team.

- Students should not be resentful towards other teammates because this can lead to miscommunication and/or using inappropriate language which can cause tension and increase conflicts.
- Students should not be a bad communicator because this can cause your team to fail a project.
- Students should not have an ego because software engineering is a team effort.

**Review by lukel43 — APPROVED** [review]
It seems to me that all the requested changes have been addressed. The content looks much better now in my opinion. Thanks so much for the hard work!

**Review by noorbuchi — APPROVED** [review]
:shipit:

**Review by mariakimheinert — APPROVED** [review]
@watson02, these changes look great! Thanks for all your hard work, Rian. :shipit:

**Pull request merged**
